domain/operation/state/task.go

```diff
@@ -0,0 +1,278 @@
+	ident := taskIdent{ID: taskID}
+
+	query := `
+SELECT o.uuid AS &taskResult.operation_uuid,
+       COALESCE(u.name, m.name) AS &taskResult.receiver,
+       oa.charm_action_key AS &taskResult.name,
+       o.summary AS &taskResult.summary,
+       o.parallel AS &taskResult.parallel,
+       o.execution_group AS &taskResult.execution_group,
+       o.enqueued_at AS &taskResult.enqueued_at,
+       ot.started_at AS &taskResult.started_at,
+       ot.completed_at AS &taskResult.completed_at,
+       osv.status AS &taskResult.status,
+       os.path AS &taskResult.path
+FROM operation_task ot
+JOIN operation o ON ot.operation_uuid = o.uuid
+JOIN operation_task_status ots ON ot.uuid = ots.task_uuid
+JOIN operation_task_status_value osv ON ots.status_id = osv.id
+LEFT JOIN operation_action oa ON o.uuid = oa.operation_uuid
+LEFT JOIN operation_unit_task out ON ot.uuid = out.task_uuid
+LEFT JOIN unit u ON out.unit_uuid = u.uuid
+LEFT JOIN operation_machine_task omt ON ot.uuid = omt.task_uuid
+LEFT JOIN machine m ON omt.machine_uuid = m.uuid
+LEFT JOIN operation_task_output oto ON ot.uuid = oto.task_uuid
+LEFT JOIN v_object_store_metadata os ON oto.store_uuid = os.uuid
+WHERE ot.task_id = $taskIdent.task_id
+`
+	stmt, err := s.Prepare(query, taskResult{}, ident)
+	if err != nil {
+		return taskResult{}, errors.Errorf("preparing get action statement: %w", err)
+	}
+
+	var result taskResult
+	err = tx.Query(ctx, stmt, ident).Get(&result)
+	if errors.Is(err, sql.ErrNoRows) {
+		return taskResult{}, errors.Errorf("action with task ID %q not found", taskID).Add(operationerrors.TaskNotFound)
+	} else if err != nil {
+		return taskResult{}, errors.Errorf("querying action with task ID %q: %w", taskID, err)
+	}
+
+	return result, nil
+}
+
+func (s *State) getOperationParameters(ctx context.Context, tx *sqlair.TX, operationUUIDStr string) ([]taskParameter, error) {
+	opUUID := uuid{UUID: operationUUIDStr}
+
+	query := `
+SELECT operation_uuid AS &taskParameter.operation_uuid,
+       key AS &taskParameter.key,
+       value AS &taskParameter.value
+FROM   operation_parameter
+WHERE  operation_uuid = $uuid.uuid
+`
+
+	stmt, err := s.Prepare(query, taskParameter{}, opUUID)
+	if err != nil {
+		return nil, errors.Errorf("preparing parameters statement: %w", err)
+	}
+
+	var parameters []taskParameter
+	err = tx.Query(ctx, stmt, opUUID).GetAll(&parameters)
+	if err != nil && !errors.Is(err, sql.ErrNoRows) {
+		return nil, errors.Errorf("querying action parameters: %w", err)
+	}
+
+	return parameters, nil
+}
+
+func (s *State) getTaskLog(ctx context.Context, tx *sqlair.TX, taskID string) ([]taskLogEntry, error) {
+	ident := taskIdent{ID: taskID}
+
+	query := `
+SELECT task_uuid AS &taskLogEntry.task_uuid,
+       content AS &taskLogEntry.content,
+       created_at AS &taskLogEntry.created_at
+FROM   operation_task_log
+JOIN   operation_task ot ON operation_task_log.task_uuid = ot.uuid
```

Comment:
```suggestion
SELECT (otl.task_uuid, otl.content, otl.created_at)AS &taskLogEntry.*
FROM   operation_task_log AS otl
JOIN   operation_task AS ot ON otl.task_uuid = ot.uuid
```

Explicit aliasing, please, not implicit [per the sql best practice](https://docs.sqlfluff.com/en/stable/reference/rules.html#bundle-aliasing)